src/Documents/DocumentStore.ts

```diff
@@ -85,14 +85,42 @@ export class DocumentStore implements IDocumentStore {
       .then((): IDocumentStore => this);
   }
 
-  public openSession(database?: string, forceReadFromMaster: boolean = false): IDocumentSession {
+  public async openSession(transaction: ITransaction) : Promise<void>;
+  public async openSession(options: ISessionOptions, transaction: ITransaction) : Promise<void>;
+  public openSession(database?: string, forceReadFromMaster?: boolean) : IDocumentSession;
+  public openSession(firstArg?: string | ITransaction | ISessionOptions, secondArg?: ITransaction | boolean): IDocumentSession | Promise<void> {
```

Comment:
@ayende type definition for callback function.
openSession callback should receive session instance and return promise